File: frontend/src/features/usergroup/http/response/UserGroupDetail.ts
import type { Username } from '@/features/user/model/Username'
import type { UserGroupDescription } from '@/features/usergroup/model/UserGroupDescription'
import type { UserGroupId } from '@/features/usergroup/model/UserGroupId'
import type { UserGroupMember } from '@/features/usergroup/model/UserGroupMember'
import type { UserGroupName } from '@/features/usergroup/model/UserGroupName'
import type { UserGroupSlug } from '@/features/usergroup/model/UserGroupSlug'
import type { AuditFields } from '@/shared/model/AuditFields'

export type UserGroupDetail = AuditFields & {
  id: UserGroupId
  slug: UserGroupSlug
  name: UserGroupName
  description: UserGroupDescription
  ownerUsername: Username
  members: UserGroupMember[]
}

File: backend/src/main/scala/domains/usergroup/model/response/UserGroupDetail.scala
package domains.usergroup.model.response

import domains.usergroup.model.*

import domains.user.model.Username

import java.time.Instant

final case class UserGroupDetail(
  id: UserGroupId,
  slug: UserGroupSlug,
  name: UserGroupName,
  description: UserGroupDescription,
  ownerUsername: Username,
  members: List[UserGroupMember],
  createdAt: Instant,
  updatedAt: Instant
)
